cts-java/src/test/java/org/ga4gh/cts/api/references/ReferenceSetsPagingIT.java

```diff
@@ -0,0 +1,132 @@
+import org.ga4gh.methods.SearchReferenceSetsRequest;
+import org.ga4gh.methods.SearchReferenceSetsResponse;
+import org.ga4gh.models.ReferenceSet;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test paging through the results returned by <tt>/references/search</tt>.
+ *
+ * @author Herb Jellinek
+ */
+@Category(DatasetsTests.class)
+public class ReferenceSetsPagingIT {
+
+    private static Client client = new Client(URLMAPPING.getInstance());
+
+    /**
+     * Check that we can page 1 by 1 through the {@link ReferenceSet}s we receive from
+     * {@link Client.References#searchReferenceSets(SearchReferenceSetsRequest)}.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkPagingOneByOneThroughReferenceSets() throws AvroRemoteException {
+
+        // retrieve them all
+        final List<ReferenceSet> listOfReferenceSets = Utils.getAllReferenceSets(client);
+        assertThat(listOfReferenceSets).isNotEmpty();
+
+        // we will remove ReferenceSets from this Set and assert at the end that we have zero
+        final Set<ReferenceSet> setOfReferenceSets = new HashSet<>(listOfReferenceSets);
+        assertThat(listOfReferenceSets).hasSize(setOfReferenceSets.size());
+
+        // page through the ReferenceSets using the same query parameters
+        String pageToken = null;
+        for (ReferenceSet ignored : listOfReferenceSets) {
+            final SearchReferenceSetsRequest pageReq =
+                    SearchReferenceSetsRequest.newBuilder()
+                                           .setPageSize(1)
+                                           .setPageToken(pageToken)
+                                           .build();
+            final SearchReferenceSetsResponse pageResp = client.references.searchReferenceSets(pageReq);
+            final List<ReferenceSet> pageOfReferenceSets = pageResp.getReferenceSets();
+            pageToken = pageResp.getNextPageToken();
+
+            assertThat(pageOfReferenceSets).hasSize(1);
+            assertThat(setOfReferenceSets).contains(pageOfReferenceSets.get(0));
+
+            setOfReferenceSets.remove(pageOfReferenceSets.get(0));
+        }
+
+        assertThat(pageToken).isNull();
+        assertThat(setOfReferenceSets).isEmpty();
+    }
+
+    /**
+     * Check that we can page through the {@link ReferenceSet}s we receive from
+     * {@link Client.References#searchReferenceSets(SearchReferenceSetsRequest)}
+     * using an increment as large as the non-paged set of results.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkPagingByOneChunkThroughReferenceSets() throws AvroRemoteException {
+
```

Comment:
This is unlikely to fail in practise, but the server is not *required* to satisfy the pageSize requests. It's perfectly entitled to send back the results in pages of 1, if it so desires. This won't fail now because we only have a handful of objects, but if there was 1000 reference sets, then some servers would send back the results in chunks smaller than 1000. As such, I think this is a bit problematic and should possibly be removed.